services/web/project/app/views.py

```diff
@@ -35,48 +41,71 @@ def new_question():
+        return redirect(url_for(
+            "views.question",
+            method="getQuestion",
+            arg1=q.id))
 
     return render_template("new_question.html", form=form)
 
 
-@views.route('/questions/<question_id>', methods=['GET', 'POST'])
-def question(question_id):
-    # Get question from DB
-    question = Question.query.get(question_id)
-    # Validate that question exists; if not, route to questions forum
-    if question is None:
-        questions()
-        questions()
-        questions()
-    # Get list of answers for question
-    question.answers = Answer.query.filter_by(questionId=question.id).all()
-    question.numAnswers = len(question.answers)
-    # Get the question's creator and assign it as an attribute
-    question.creator = User.query.get(question.userId)
-
-    # Process form
-    form = NewAnswerForm()
-    if form.validate_on_submit():
-        body = form.body.data
-        # Add answer to DB
-        a = Answer(body, current_user.id, question.id)
-        db.session.add(a)
-        db.session.commit()
-        return redirect(url_for('views.question', question_id=question_id))
+@views.route('/questions/', defaults={
+    'method': 'None',
+    'arg1': 'None',
+    'arg2': 'None'})
+@views.route('/questions/<method>/', defaults={
+    'arg1': 'None',
+    'arg2': 'None'})
+@views.route('/questions/<method>/<arg1>', defaults={
+    'arg2': 'None'},
+    methods=['GET', 'POST'])
+@views.route('/questions/<method>/<arg1>/<arg2>', methods=['GET', 'POST'])
+def question(method, arg1, arg2):
+    # Make best answer method
+    if(method == "acceptAnswer"):
+        result = QuestionCtrl.acceptAnswer(arg1, arg2)
+        if(result[0] == "REDIRECT"):
+            return redirect(url_for(
+                result[1],
+                method=result[2],
+                arg1=result[3]))
+        else:
+            return result
+    # Get Queston method
+    elif(method == "getQuestion"):
+        result = QuestionCtrl.getQuestion(arg1)
+        if(result[0] == "REDIRECT_1"):
+            return redirect(url_for(result[1]))
+        elif(result[0] == "REDIRECT_2"):
+            return redirect(url_for(
+                result[1],
+                method=result[2],
+                arg1=result[3]))
+        else:
+            return render_template(
+                result[1],
+                question=result[2],
+                form=result[3])
+    else:
+        return redirect(url_for("views.questions"))
 
-    # For each answer, add the creator as an attribute
-    for a in question.answers:
-        a.creator = User.query.get(a.userId)
-    return render_template('question.html', question=question, form=form)
 
-# BUG: The function below does not return the files in static as expected.
-# @views.route("/static/<path:filename>")
-# def static_file(filename):
-#     return send_from_directory(app.config["STATIC_FOLDER"], filename)
+@views.route('/vote/<question_id>/<answer_id>/<value>', methods=['GET'])
```

Comment:
This if you we follow REST should be post or patch

Reply:
Sure, I'll change it to POST.